src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -331,10 +333,11 @@ protected void retrieve(SCMSourceCriteria criteria, @NonNull SCMHeadObserver obs
                         boolean fork = !m.getSourceProjectId().equals(m.getTargetProjectId());
                         String originOwner = m.getAuthor().getUsername();
                         String originProjectPath = projectPath;
-                        if (fork) {
+                        if (fork && !mrSources.containsKey(mr.getSourceProjectId())) {
```

Comment:
currently, it doesn't seem to retrieve the originProjectPath if `mrSources` already contains the project id :)

Reply:
Fixed in cb980a0 :)